Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/SelectionComponent.cpp

```diff
@@ -5,90 +5,84 @@
  * SPDX-License-Identifier: Apache-2.0 OR MIT
  *
  */
+
 #include "SelectionComponent.h"
-#include <AzCore/Serialization/EditContext.h>
-#include <AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI.h>
+
+#include <AzCore/Memory/Memory.h>
+#include <AzCore/Serialization/Json/BaseJsonSerializer.h>
+#include <AzCore/Serialization/Json/RegistrationContext.h>
+#include <AzToolsFramework/Viewport/ViewportSettings.h>
 
 namespace AzToolsFramework
 {
     namespace Components
     {
-        SelectionComponent::SelectionComponent()
-        {
-            m_currentSelectionFlag = SF_Unselected;
-            m_currentSelectionAABB = AZ::Aabb::CreateNull();
-        }
+        static constexpr const char* const SelectionComponentRemovalNotice =
+            "[INFORMATION] The editor SelectionComponent is being removed from .prefab data. Please edit and save the level to persist "
```

Comment:
nit: The notice to 'edit' the level may not be necessary here. When the level with a selection component is loaded, it should be marked dirty forcing the user to save the level before exiting right? So the user doesn't need to do explicit edits to be able to save the level.

Reply:
So in my experience when testing this the level actually wasn't flagged as dirty after this happened 🙁 So if you immediately closed the level (without making any changes) and opened it again, you'd get the same spam (until you modified the level and explicitly saved - which is why I chose to include this text to make it clear). So potentially something in the Prefab system isn't working as expected? 🤔